### PR TITLE
Docs - Split Developing collections page, add info on optional module_utils

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY

The current page on [Developing collections](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html) is very long. This PR splits it into a section.

It also adds a paragraph about using optional module utilities with try/except blocks so collections can support multiple versions of ansible/ansible-core.

Related to #73832 and #73423.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs.ansible.com
collections

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_collections_specifics.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+   except ImportError:
+       respawn_module = None
+   ...
+   if needs_respawn:
+       if respawn_module:
+           respawn_module(target)
+       else:
+           module.fail_json('respawn is not available in Ansible < 2.11, ensure that foopkg is installed')
+
+The optional module_utils behavior also applies to module imports from collections.
```
Comment (nitzmahone) [inline docs/docsite/rst/dev_guide/developing_collections_specifics.rst]: Reading this back, it could be incorrectly interpreted that we're allowing imports *of* Ansible modules from collections. Maybe reword to something like:

```suggestion
The optional import behavior also applies to module_utils imported from collections.
```

**Review by nitzmahone — DISMISSED** [review]
Split LGTM at a glance, as does the changed intro paragraph for the optional module_utils. One minor clarification probably needed on the module_utils stuff, but otherwise +1 from me.

File: docs/docsite/rst/dev_guide/developing_collections_specifics.rst (hunk 1 above)
Comment (acozine) [reply docs/docsite/rst/dev_guide/developing_collections_specifics.rst]: Thanks @nitzmahone for the review. I changed the line number in a different edit, so the suggestion is "outdated" and I can't commit it from here. I implemented this exact change in 4b03edcf8d8ccdc1d480e86e457c70141e051c9a.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 2]
```diff
@@ -1,825 +1,48 @@
-
 .. _developing_collections:
 
 **********************
 Developing collections
 **********************
 
-Collections are a distribution format for Ansible content. You can package and distribute playbooks, roles, modules, and plugins using collections.
+Collections are a distribution format for Ansible content. You can package and distribute playbooks, roles, modules, and plugins using collections. A typical collection addresses a set of related use cases. For example, the <need example here ???????> collection automates <complete example here ?????>.
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
Collections are a distribution format for Ansible content. You can package and distribute playbooks, roles, modules, and plugins using collections. A typical collection addresses a set of related use cases. For example, the ``cisco.ios`` collection automates management of Cisco IOS devices.
```

File: docs/docsite/rst/dev_guide/developing_collections_contributing.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.. _hacking_collections:
+
+***************************
+Contributing to collections
+***************************
+
+If you want to add functionality to an existing collection, modify a collection you are using to fix a bug, or change the behavior of a module in a collection, clone the git repository for that collection and make changes on a branch. You can combine changes to a collection with a local checkout of Ansible (``source hacking/env-setup``).
+
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_collections_contributing.rst]:
```suggestion
You should first check the collection repository to see if it has specific contribution guidelines. These are typically listed in the README.md or CONTRIBUTING.md files within the repository.
```

File: docs/docsite/rst/dev_guide/developing_collections_distributing.rst [hunk 4]
```diff
@@ -0,0 +1,241 @@
+.. _distributing_collections:
+
+************************
+Distributing collections
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_collections_distributing.rst:4]: Is distributing the right term here? I thought it was publishing as the activity a collection owner does. Just wondering if someon would search 'how do I distribute my collection' vs 'how do I publish my collection' ? I also don't see anything in official AH documentation or marketing details that calls it a distribution server?

File: docs/docsite/rst/dev_guide/developing_collections_migrating.rst [hunk 5]
```diff
@@ -0,0 +1,136 @@
+.. _migrate_to_collection:
+
+***************************************************
+Migrating Ansible content to a different collection
+***************************************************
+
+When you move content from one collection to another, for example to extract a set of related modules out of community.general to create a more focused collection, you must make sure the transition is easy for users to follow.
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_collections_migrating.rst]:
```suggestion
When you move content from one collection to another, for example to extract a set of related modules out of ``community.general`` to create a more focused collection, you must make sure the transition is easy for users to follow. 
```

File: docs/docsite/rst/dev_guide/developing_collections_distributing.rst (hunk 4 above)
Comment (acozine) [reply docs/docsite/rst/dev_guide/developing_collections_distributing.rst:4]: Yeah, I debated about that. Current thinking is that "distributing" includes both creating the tarball and publishing the tarball to Galaxy/AH. I made sure the first paragraph includes both terms. Looking at it again, though, I see that I don't mention packaging it up to create the tarball . . . I'll add a sentence about that.

Comment (acozine) [reply docs/docsite/rst/dev_guide/developing_collections_distributing.rst:4]: oh, it's there, in the "three major steps" section . . . phew

**Comment by acozine** [comment]
Rebased and ready for final review.

**Review by samccann — APPROVED** [review]
